### ISSUE TITLE
Fail gracefully on App Error (missing history key in API response)

### ISSUE THREAD

**Issue by rmanny** [RoadieHQ/roadie-backstage-plugins#187]
I have an app deployed in three clusters, one of which I just removed. This means the app was properly selected by the selector key, but did not have history attached. The API code errored out because of this.

## Expected Behavior

If one of the apps is broken (Status=Unknown), this should not cause the plugin to crash and show nothing. Ideally, it should show the working apps, and then translate the broken app into a user-readable status.

## Current Behavior

The API was returning something like this (notice there is no history key).

```
{
   "metadata":{
      "resourceVersion":...
   },
   "items":[
      ...
      {
         "metadata":{
            ...
         },
         "spec":{
            ...
         },
         "status":{
            "sync":{
               "status":"Unknown",
               ...
            },
            "health":{
               "status":"Healthy"
            },
            "conditions":[
               {
                  "type":"ComparisonError",
                  "message":"rpc error: code = Unknown desc = <cluster>: app path does not exist",
                  "lastTransitionTime":"2021-06-21T21:58:53Z"
               }
            ],
            ...
         }
      }
   ]
}
```

This IS a valid Argo app, even though it is erroring.

This caused the ArgoCD overview card to display this:
<img width="293" alt="argocd overview" src="https://user-images.githubusercontent.com/77029610/122835513-560c1000-d2a5-11eb-8446-9b8cf2423bc5.PNG">
even though there were two working apps.

Here is the error text:
```
Error occurred while fetching data. Error: remote data validation failed: Expecting Array<{ id: number, revision: string, deployStartedAt: (string | undefined), deployedAt: (string | undefined) }> at items.2.status.history but instead got: undefined; Expecting { finishedAt: string } at items.2.status.operationState but instead got: undefined
```

## Possible Solution

I have not looked at the code, but a try catch around the validation inside this loop could output a simple app error status to be shown to the user.

## Steps to Reproduce

1. Create a broken ArgoCD app which is not pointing at a valid GitHub repo path
2. Set up a selector in catalog-info.yaml which selects this broken app
3. Observe that the EntityPage now shows an error instead of the expected app list

## Context

It was easy to fix the error by totally deleting the broken app, but this error seems relatively simple to fix and would save future Backstage devs the trouble.

## Your Environment

Node 14, latest version of Backstage (just ran versions:bump this morning).

**Comment by dtuite** [comment]
@Irma12 should this be closed now because https://github.com/RoadieHQ/roadie-backstage-plugins/pull/211 is merged?

**Comment by Irma12** [comment]
@rmanny Thanks for reporting the bug, it has been addressed in v1.0.13 so you can check it out 🎉
